### PR TITLE
Update to beta2 public parameters

### PR DESCRIPTION
**Pull request by ebfull**
Also gets rid of the `testnet3`/`regtest` subdirectories/symlinks and other such things.

### CODE REVIEW

**Comment by bitcartel** [comment]
utACK

File: src/util.cpp [hunk 1]
```diff
@@ -478,7 +478,6 @@ const boost::filesystem::path &ZC_GetParamsDir()
         return path;
 
     path = ZC_GetBaseParamsDir();
-    path /= BaseParams().DataDir();
```
Comment (str4d) [inline src/util.cpp]: Heh, I didn't realise this was a default overload in Boost.

File: zcutil/fetch-params.sh [hunk 2]
```diff
@@ -4,19 +4,16 @@ set -eu
...
 
-REGTEST_PKEY_NAME='z9-proving.key'
-REGTEST_VKEY_NAME='z9-verifying.key'
-REGTEST_PKEY_URL="https://z.cash/downloads/$REGTEST_PKEY_NAME"
-REGTEST_VKEY_URL="https://z.cash/downloads/$REGTEST_VKEY_NAME"
-REGTEST_DIR="$PARAMS_DIR/regtest"
-
-# This should have the same params as regtest. We use symlinks for now.
-TESTNET3_DIR="$PARAMS_DIR/testnet3"
+BETA2_PKEY_NAME='beta2-proving.key'
```
Comment (str4d) [inline zcutil/fetch-params.sh]: Non-blocking: These could just be named `PKEY_NAME` etc. to reduce size of future diffs.

File: zcutil/fetch-params.sh [hunk 3]
```diff
@@ -86,30 +93,10 @@ $README_PATH
...
-    if [ $CHECKSUM_RESULT -eq 0 ]; then
-        echo 'Updating testnet3 symlinks to regtest parameters.'
-        mkdir -p "$TESTNET3_DIR"
-        ln -sf "../regtest/$REGTEST_PKEY_NAME" "$TESTNET3_DIR/$REGTEST_PKEY_NAME"
-        ln -sf "../regtest/$REGTEST_VKEY_NAME" "$TESTNET3_DIR/$REGTEST_VKEY_NAME"
-    else
-       exit 1
-    fi
-
+    fetch_params "$BETA2_PKEY_URL" "$PARAMS_DIR/$BETA2_PKEY_NAME" "cca9887becf803c8ca801bc9da8fcba4f5fb3ba13af9d17e8603021a150cb4b7"
```
Comment (str4d) [inline zcutil/fetch-params.sh]: I'd put these in variables up next to `PKEY_NAME` etc, again to reduce the size of future diffs.

File: zcutil/fetch-params.sh [hunk 4]
```diff
@@ -27,8 +24,18 @@ function fetch_params {
...
 
-        # Only after successful download do we update the parameter load path:
-        mv -v "$dlname" "$output"
+        shasum -a 256 --check <<EOF
+$expectedhash  $dlname
+EOF
+
+        # Check the exit code of the shasum command:
+        CHECKSUM_RESULT=$?
+        if [ $CHECKSUM_RESULT -eq 0 ]; then
```
Comment (str4d) [inline zcutil/fetch-params.sh]: Does this need to be quoted, or is `$?` guaranteed to never return an empty string?

File: src/util.cpp (hunk 1 above)
Comment (ebfull) [inline src/util.cpp]: I really dislike APIs that hijack operators like that...

**Comment by str4d** [comment]
Suggested changes above can be done later.

@zkbot r+

**Comment by str4d** [comment]
Fixed the bug in #1429 so this should work this time.

@zkbot retry

**Pull request merged**